README.md

```diff
@@ -81,6 +81,29 @@ Currently the component that takes the longest time to compile is
 `ihaskell-widgets`, so if you're in a hurry you may want to comment that out in
 `stack.yaml`.
 
+## Stack development with Docker
+This is an alternative way to use Docker than above, taking advantage of stack's Docker support.
+Makes it easy to install new packages with `stack --docker install <pkg>` without having to rebuild a docker image.
+Like the other Docker workflow, this doesn't require any host dependecies to be installed.
+
+```
+docker build -t ihaskell-dev docker
+stack --docker setup
+stacl --docker install
+stack --docker exec ihaskell -- install --stack
+stack --docker exec jupyter -- notebook --ip=0.0.0.0 notebooks
```

Comment:
Can we move some of these instructions out of the `README` and into the `Dockerfile`? I think everything up until the final `stack --docker exec jupyter -- notebook --ip=0.0.0.0 notebooks` should be safe to put in the `Dockerfile`. If not, why?

Reply:
The docker image used doesn't actually contain the IHaskell source, so `stack` can't see the `stack.yaml` which is outside the docker build context, can it?

All the build artifacts and packages will be on the _host_ (volume mounted into the docker container), which allows installing additional packages without rebuilding the docker image and keeping the docker container ephemeral.

Reply:
You're right, I had forgotten about this.